src/agent/debuglet.ts

```diff
@@ -106,6 +112,46 @@ const formatBreakpoints = function(
           .join('\n');
 };
 
+/**
+ * CachedPromise class is designed to support debug agent in Google Cloud
+ * Function (GCF). As GCF is a serverless environment and we wanted to make
+ * sure debug agent could always capture the snapshots. We designed this class
+ * so that GCF could wait the ListActiveBreakpoitns event completed before
+ * throttling CPU to 0. CachedPromise has two member functions.
+ * get() is to create or return an existing promise. resolve() will resolve
+ * the promise at appropriate code location (in our case is after
+ * listActiveBreakpoints executed).
```

Comment:
More nit: listActiveBreakpoints *is* executed

BTW is it *l*istActiveBreakpoints or *L*istActiveBreakpoints?

Reply:
Done. it is listActiveBreakpoints.